etc/calico-diags

```diff
@@ -77,11 +77,18 @@ fi
 cp -a "$CALICO_CFG" etc_calico
 
 # Use grep -v to attempt to strip out password information
-grep -v -i "password" /etc/nova/nova.conf > nova.conf 2>&1
-grep -v -i "password" /etc/neutron/neutron.conf > neutron.conf 2>&1
-grep -v -i "password" /etc/neutron/plugins/ml2/ml2_conf.ini > ml2_conf.ini 2>&1
-grep -v -i "password" /etc/neutron/dhcp_agent.ini > dhcp_agent.ini 2>&1
-grep -v -i "password" /etc/neutron/metadata_agent.ini > metadata_agent.ini 2>&1
+function copy_and_redact() {
+  in=$1
+  out=$2
+
+  sed -e 's/^\([ \t]*\(password\|token\|transport_url\|connection\)[ \t]*=[ \t]*\).*/\1REDACTED/' "$in" > $out 2>&1
```

Comment:
Wondering about the initial `^` here - is it needed?

Without it, this expression would also catch - for example - `auth_token = xxx`.  But as it stands, I believe it would miss that.